packages/cli/src/utils/package_manager.ts

```diff
@@ -0,0 +1,95 @@
+import { exec } from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as clack from '@clack/prompts'
+import { PackageManager } from '../types'
+import { abort, abortIfCancelled } from './clack_utils'
+import chalk from 'chalk'
+import { setTimeout as sleep } from 'node:timers/promises'
+
+const yarn: PackageManager = {
+  name: 'yarn',
+  label: 'Yarn',
+  lockFile: 'yarn.lock',
+  installCommand: 'yarn add',
+  buildCommand: 'yarn build',
+}
+const pnpm: PackageManager = {
+  name: 'pnpm',
+  label: 'PNPM',
+  lockFile: 'pnpm-lock.yaml',
+  installCommand: 'pnpm install',
+  buildCommand: 'pnpm build',
+}
+const npm: PackageManager = {
+  name: 'npm',
+  label: 'NPM',
+  lockFile: 'package-lock.json',
+  installCommand: 'npm install',
+  buildCommand: 'npm run build',
+}
+
+export const packageManagers = [yarn, pnpm, npm]
+
+export function detectPackageManger(): PackageManager | null {
+  for (const packageManager of packageManagers) {
+    if (fs.existsSync(path.join(process.cwd(), packageManager.lockFile))) {
+      return packageManager
+    }
+  }
+  return null
+}
+
+export async function installPackageWithPackageManager(
+  packageManager: PackageManager,
+  packageName: string,
+): Promise<void> {
+  await exec(`${packageManager.installCommand} ${packageName}`)
+}
+
+async function getPackageManager(): Promise<PackageManager> {
+  const s = clack.spinner()
+  s.start('Detecting package manager.')
+
+  const detectedPackageManager = detectPackageManger()
+  await sleep(1000)
+  s.stop(chalk.bold(detectedPackageManager?.label || 'Nothing') + ' detected.')
+
+  if (detectedPackageManager) {
+    return detectedPackageManager
+  }
+
+  const selectedPackageManager: PackageManager | symbol = await abortIfCancelled(
```

Comment:
You can use `@antfu/ni` to detect package manager (e.g. https://github.com/shadcn-ui/ui/blob/c21ecfb665214e18cd5914ea319f925cd676e786/packages/cli/src/utils/get-package-manager.ts#L4)

Reply:
Yes I can but here I need install command too. Also in the backend antfu/ni also uses similar process to detect it https://github.com/antfu/ni/blob/main/src/detect.ts. So dont feel like adding extra dependency.